src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -663,39 +641,13 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
         }
     }
 
-    void Schedule::applyWRFT(const DeckKeyword& keyword, std::size_t currentStep) {
-        /* Rule for handling RFT: Request current RFT data output for specified wells, plus output when
-         * any well is subsequently opened
-         */
-
-        for (const auto& record : keyword) {
 
-            const std::string& wellNamePattern = record.getItem("WELL").getTrimmedString(0);
-            const auto well_names = wellNames(wellNamePattern, currentStep);
-            for (const auto& well_name : well_names)
-                this->rft_config.updateRFT(well_name, currentStep, RFTConfig::RFT::YES);
-
-        }
-
-        this->rft_config.setWellOpenRFT(currentStep);
-    }
-
-    void Schedule::applyWRFTPLT(const DeckKeyword& keyword, std::size_t currentStep) {
-        for (const auto& record : keyword) {
-            const std::string& wellNamePattern = record.getItem("WELL").getTrimmedString(0);
-
-            RFTConfig::RFT RFTKey = RFTConfig::RFTFromString(record.getItem("OUTPUT_RFT").getTrimmedString(0));
-            RFTConfig::PLT PLTKey = RFTConfig::PLTFromString(record.getItem("OUTPUT_PLT").getTrimmedString(0));
-            const auto well_names = wellNames(wellNamePattern, currentStep);
-            for (const auto& well_name : well_names) {
-                this->rft_config.updateRFT(well_name, currentStep, RFTKey);
-                this->rft_config.updatePLT(well_name, currentStep, PLTKey);
-            }
+    std::optional<std::size_t> Schedule::first_RFT() const {
+        for (std::size_t report_step = 0; report_step < this->snapshots.size(); report_step++) {
+            if (this->snapshots[report_step].rft_config().active())
+                return report_step;
```

Comment:
Just for the record this predicate used to be constant time.  Now it's linear in the number of report steps.

Reply:
Yes  you are right ;-)